### PR TITLE
feat(staking): load multi-delegation for HW wallets

### PR DESCRIPTION
**Pull request by xdzurman**
# Checklist

- [x] JIRA - \<link>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Allow multi-delegation for HW wallets. This PR also fixes broken "back" arrow button (also button was rendered where it wasn't supposed to - preferences, success, failure, I fixed that as well). This functionality is implementeed via a `skip` step, similarly to pre-machine implementation. After submitting a HW staking tx, you go straight into success or failure.

## Testing

Use a hardware wallet (ledger) to test out all multi-delegation functionality. Update the cardano app according to the guide here https://input-output.atlassian.net/browse/LW-8937

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2811/6730657570/index.html) for [40e4053b](https://github.com/input-output-hk/lace/pull/686/commits/40e4053be724ee06aa205f053d770cdc1b46886d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 22     | 9      | 0       | 0     | 31    | ❌     |

### CODE REVIEW

File: packages/staking/src/features/store/delegationPortfolioStore/stateMachine/commands.ts [hunk 1]
```diff
@@ -117,10 +125,16 @@ export type PortfolioManagementPreferencesCommand =
...
+export type PortfolioManagementConfirmationCommand =
+  | CancelDrawer
+  | DrawerContinue
+  | DrawerBack
+  | HwSkipToSuccess
+  | HwSkipToFailure;
 
 export type PortfolioManagementSignCommand = CancelDrawer | DrawerContinue | DrawerFailure | DrawerBack;
 
+// TODO: should not have DrawerBack, it's not used
```
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/store/delegationPortfolioStore/stateMachine/commands.ts:137]: 👍🏻 please resolve when PR isn't WIP

File: packages/staking/src/features/Drawer/StakePoolDetailsDrawer.tsx [hunk 2]
```diff
@@ -72,17 +72,20 @@ export const StakePoolDetailsDrawer = ({
...
     } else {
       closeDrawer();
     }
   });
 
   const createArrowIconCallback = () => {
-    if (activeDrawerStep && typeof showBackIcon === 'function' ? showBackIcon(activeDrawerStep) : showBackIcon) {
-      return popupView ? closeDrawer : onGoBack;
+    if (shouldShowBackIcon) {
+      return popupView ? closeDrawer() : onGoBack();
```
[the remark this answers is not in the document]
Comment (xdzurman) [reply packages/staking/src/features/Drawer/StakePoolDetailsDrawer.tsx:88]: It does not work without it. This is a fix

[the remark this answers is not in the document]
Comment (przemyslaw-wlodek) [reply packages/staking/src/features/Drawer/StakePoolDetailsDrawer.tsx:88]: Sorry, you're right -> I was reviewing this in a rush. Then the name of the function is misleading as it doesn't create a callback, it is a callback 😅

[the remark this answers is not in the document]
Comment (xdzurman) [reply packages/staking/src/features/Drawer/StakePoolDetailsDrawer.tsx:88]: I'll rename it

**Review by przemyslaw-wlodek — APPROVED** [review]
Can't wait for the reactions on the new releases. 😄

**Comment by refi93** [comment]
I guess for Trezor multidelegation hasn't been officiallt released yet, probably neither for Ledger (latest available version I found in Ledger Live is 5.0.1) and if that's the case I think we should add separate feature flags for enabling multidelegation for trezor and ledger wallets to ensure the feature doesn't become live before hw wallets actually support this, or am I missing something?

https://trezor.io/learn/a/firmware-changelog

https://github.com/vacuumlabs/ledger-app-cardano-shelley/commits/develop -> multidelegation available in Ledget Cardano app 6.1.2 (not released yet, apparently)

**Review by refi93 — CHANGES_REQUESTED** [review]
I believe a feature flag to gate the feature in production is probably needed, see comment

**Review by refi93 — APPROVED** [review]

**Review by oldGreg5 — APPROVED** [review]

**Pull request merged**